src/resource-archive/index.ts

```diff
@@ -68,8 +97,22 @@ class Watcher {
     method: T,
     params?: Protocol.CommandParameters[T]
   ): Promise<Protocol.CommandReturnValues[T] | null> {
+    let resolver: any = null;
+    const prom = new Promise((resolve, reject) => {
+      resolver = resolve;
+    });
```

Comment:
Probably worth pulling this out to a helper to make it clearer, maybe something like:

```ts
const complete = this.registerUnfulfilledRequest(request.url);

// later
complete();
```

Then the `registerUnfulfilledRequest` could look something like:

```ts
registerUnfulfilledRequest() {
  const complete = // something similar to what you have
 
  complete.then(() => {
     if (/* all requests are resolved */) {
       this.allRequestsDoneComplete(); // this triggers a "super" promise representing all requests
     }
   }

  return complete;
}
```